src/main/java/fi/otavanopisto/kuntaapi/server/integrations/management/ManagementTileEntityUpdater.java

```diff
@@ -97,90 +112,138 @@ public void stopTimer() {
-    for (Tile managementTile : managementTiles) {
-      updateManagementTile(organizationId, api, managementTile);
-    }
-  }
-
-  private List<Tile> listManagementTiles(DefaultApi api, OrganizationId organizationId) {
-    fi.metatavu.management.client.ApiResponse<List<Tile>> response = api.wpV2TileGet(null, null, null, null, null, null, null, null, null, null, null, null, null, null);
+  private void updateManagementTile(DefaultApi api, OrganizationId organizationId, TileId managementTileId) {
+    ApiResponse<Tile> response = api.wpV2TileIdGet(managementTileId.getId(), null);
     if (response.isOk()) {
-      return response.getResponse();
+      updateManagementTile(api, organizationId, response.getResponse());
     } else {
-      logger.warning(String.format("Listing organization %s tiles failed on [%d] %s", organizationId.getId(), response.getStatus(), response.getMessage()));
+      logger.warning(String.format("Finding organization %s tile failed on [%d] %s", managementTileId.getId(), response.getStatus(), response.getMessage()));
     }
-    
-    return Collections.emptyList();
   }
   
-  private void updateManagementTile(OrganizationId organizationId, DefaultApi api, Tile managementTile) {
+  private void updateManagementTile(DefaultApi api, OrganizationId organizationId, Tile managementTile) {
     TileId tileId = new TileId(organizationId, ManagementConsts.IDENTIFIER_NAME, String.valueOf(managementTile.getId()));
 
     Identifier identifier = identifierController.findIdentifierById(tileId);
     if (identifier == null) {
       identifier = identifierController.createIdentifier(tileId);
     }
     
-    modificationHashCache.put(identifier.getKuntaApiId(), createPojoHash(managementTile));
+    TileId kuntaApiTileId = new TileId(organizationId, KuntaApiConsts.IDENTIFIER_NAME, identifier.getKuntaApiId());
+    fi.otavanopisto.kuntaapi.server.rest.model.Tile tile = managementTranslator.translateTile(kuntaApiTileId, managementTile);
+    if (tile == null) {
+      logger.severe(String.format("Could not translate management tile %s", identifier.getKuntaApiId()));
+      return;
+    }
+    
+    tileCache.put(kuntaApiTileId, tile);
+    modificationHashCache.put(identifier.getKuntaApiId(), createPojoHash(tile));
     
     if (managementTile.getFeaturedMedia() != null && managementTile.getFeaturedMedia() > 0) {
-      updateFeaturedMedia(organizationId, api, managementTile.getFeaturedMedia()); 
+      updateFeaturedMedia(organizationId, kuntaApiTileId, api, managementTile.getFeaturedMedia()); 
     }
   }
   
-  private void updateFeaturedMedia(OrganizationId organizationId, DefaultApi api, Integer featuredMedia) {
-    ApiResponse<fi.metatavu.management.client.model.Attachment> response = api.wpV2MediaIdGet(String.valueOf(featuredMedia), null);
+  private void updateFeaturedMedia(OrganizationId organizationId, TileId kuntaApiId, DefaultApi api, Integer featuredMedia) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "updateFeaturedMedia" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)